pymodbus/transport/transport.py

```diff
@@ -289,7 +304,7 @@ def connection_made(self, transport: asyncio.BaseTransport):
         self.reset_delay()
         self.callback_connected()
 
-    def connection_lost(self, reason: Exception):
+    def connection_lost(self, reason: Exception | None) -> None:
```

Comment:
Can that be called with None ?

Reply:
See comment 11.

Reply:
As MAKOMO explained, the default is in fact calling with `None` (via a lambda)

https://github.com/pymodbus-dev/pymodbus/blob/2adff1fe5564526e3fd742de5b05e1c8790356f8/pymodbus/transport/transport.py#L162